libvips/foreign/heifsave.c

```diff
@@ -293,7 +334,7 @@ vips_foreign_save_heif_write_page(VipsForeignSaveHeif *heif, int page)
 		}
 	}
 
-	if (!save->strip &&
+	if (save->preserve &&
 		vips_foreign_save_heif_write_metadata(heif))
```

Comment:
You're right, I think this check was needed to prevent `vips__exif_update()` from being called when `preserve=none`. In that case, the entire EXIF ​​block does not need to be rebuilt.

Fixed with commit 41bcdefb94dcb2080c065ee8ee088905a90aad16.